consumer/consumererror/permanent.go

```diff
@@ -14,27 +14,60 @@
 
 package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
 
-import "errors"
+import (
+	"errors"
+	"strconv"
+)
 
-// permanent is an error that will be always returned if its source
+// NO_DATA_COUNT indicates an unknown quantity of telemetry items.
+const NO_DATA_COUNT = -1
+
+// Permanent is an error that will always be returned if its source
 // receives the same inputs.
-type permanent struct {
-	err error
+type Permanent struct {
+	error
+	successful int
```

Comment:
I added this since I think it would provide good contextual information for the failed batch.

I think `total` could also work here, but `successful` seems like a better match for a `failed` count.

Reply:
Yeah I agree. With total you can derive "successful"

Reply:
It looks like the successful records are calculated by the `obsreport` package and the totals are probably best left to be recorded by the caller, so I'm going to leave this out for now.